pkg/client/testclient/fake.go

```diff
@@ -9,19 +9,10 @@ import (
 	"github.com/openshift/origin/pkg/client"
 )
 
-// FakeAction is a wrapper around the Kubernetes FakeAction.
-// Used for faking various actions on resources.
-type FakeAction ktestclient.FakeAction
-
 // Fake implements Interface. Meant to be embedded into a struct to get a default
 // implementation. This makes faking out just the method you want to test easier.
 type Fake struct {
-	// Fake by default keeps a simple list of the methods that have been called.
-	Actions []FakeAction
-	Err     error
-	// ReactFn is an optional function that will be invoked with the provided action
-	// and return a response.
-	ReactFn ktestclient.ReactionFunc
+	*ktestclient.Fake
```

Comment:
Hmm… if it's a huge change it probably isn't worth it. Let's think if
there's a cleaner way to unify the actions. Could we time stamp them and
combine/sort them?


On Jul 8, 2015, at 7:32 AM, Michail Kargakis <notifications@github.com>
wrote:

In pkg/client/testclient/fake.go
<https://github.com/openshift/origin/pull/3601#discussion_r34138075>:

>  // Fake implements Interface. Meant to be embedded into a struct to get a default
>  // implementation. This makes faking out just the method you want to test easier.
>  type Fake struct {
> -	// Fake by default keeps a simple list of the methods that have been called.
> -	Actions []FakeAction
> -	Err     error
> -	// ReactFn is an optional function that will be invoked with the provided action
> -	// and return a response.
> -	ReactFn ktestclient.ReactionFunc
> +	*ktestclient.Fake

Hmm, this is so involved ... And I am still fixing tests in OpenShift,
still havent moved upstream

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/3601/files#r34138075>.